app/Phragile/BurndownChart.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace Phragile;
+
+class BurndownChart {
+	// TODO: ideally these should come from a cached call to maniphest.querystatuses
+	private static $STATUS_OPEN = ['stalled', 'open'];
+
+	private $pointsClosedBeforeSprint = null;
+	private $pointsClosedPerDay = null;
+	private $sprint = null;
+	private $tasks = null;
+	private $phabricator = null;
+
+	public function __construct(\Sprint $sprint, TaskList $tasks, PhabricatorAPI $phabricator)
+	{
+		$this->sprint = $sprint;
+		$this->tasks = $tasks;
+		$this->phabricator = $phabricator;
+	}
+
+	private function calculatePointsClosedPerDay()
+	{
+		$dateFormat = 'Y-m-d';
+		$sprintDuration = $this->sprint->formatDays($dateFormat);
+		$this->pointsClosedPerDay = array_fill_keys($sprintDuration, 0);
+
+		foreach ($this->getClosedTaskTimes() as $id => $time)
+		{
+			$taskClosedDay = date($dateFormat, $time);
+
+			if (in_array($taskClosedDay, $sprintDuration))
+			{
+				$this->pointsClosedPerDay[$taskClosedDay] += $this->tasks->findTaskByID($id)['story_points'];
+			} elseif ($taskClosedDay < $this->sprint->sprint_start)
+			{
+				$this->pointsClosedBeforeSprint += $this->tasks->findTaskByID($id)['story_points'];
+			}
+		}
+	}
+
+	private function getClosedTaskIDs()
+	{
+		return array_map(function($task)
+		{
+			return $task['id'];
+		}, array_filter($this->tasks->getTasks(), function($task)
+			{
+				return $task['closed'];
+			})
+		);
+	}
+
+	private function getClosedTaskTimes()
+	{
+		$taskTransactions = $this->phabricator->taskTransactions($this->getClosedTaskIDs());
+
+		return array_map(function($transactions)
+		{
+			return $this->findLastStatusChangeToClosed($transactions);
+		}, $taskTransactions);
+	}
+
+	private function findLastStatusChangeToClosed(array $transactions)
+	{
+		return array_reduce(
+			$transactions,
+			function($time, $transaction)
+			{
+				if ($transaction['transactionType'] === 'status'
```

Comment:
You can extract this predicate to its own function that takes the $transaction as argument. Then the function name will make the code easier to read. Something like `statusChangedFromOpen`, if that indeed is what this if statement is about.